### ISSUE TITLE
check if basic block child of correct type

### ISSUE THREAD

**Issue by mike-hunhoff** [fireeye/capa#494]
we should add a check to the following code that verifies the single child statement is of the expected type:

https://github.com/fireeye/capa/blob/14e65c46011aca908863a770ec925cc222dc7f93/capa/rules.py#L347-L354

**Comment by mike-hunhoff** [comment]
from my understanding a single subscope child statement must be one of `and`, `or`, `optional`, `not`, `N or more`?

however, it looks like we have a rule (maybe more I haven't verified) that have gotten away with a single subscope child statement of any type e.g.

https://github.com/fireeye/capa-rules/blob/1ff994f7916d66e39b4b5b8dbb310d0e0b051f7f/compiler/rust/compiled-with-rust.yml#L1-L14

do we need to restrict the type of the single subscope child statement? I feel like this is the intention, but capa doesn't enforce this. IMO we should enforce the type to avoid confusion e.g. if I can nest one `mnemonic` under `basic block` why can't I nest two?

![Screen Shot 2021-03-26 at 9 02 19 AM](https://user-images.githubusercontent.com/42192796/112651648-38e4b400-8e12-11eb-9445-fae963545ede.png)

![Screen Shot 2021-03-26 at 9 02 34 AM](https://user-images.githubusercontent.com/42192796/112651664-3c783b00-8e12-11eb-8469-80cacacdd7fe.png)

thoughts @Ana06 , @mr-tz , @williballenthin ?

**Comment by williballenthin** [comment]
yes, the subscope should have a single child item, one of and/or/X or more.

a subscope block is translated (behind the scenes) into a separate capa rule and all capa rules should have a single root element. therefore, we should enforce the same at each subscope.

**Comment by williballenthin** [comment]
looks like this is only enforced during deserialization, though we should probably have a lint for it, too

https://github.com/fireeye/capa/blob/master/capa/rules.py#L565-L568
![image](https://user-images.githubusercontent.com/156560/112654177-bd383680-8e14-11eb-9451-2caa44b1f29e.png)
